### PR TITLE
Wavelet (inter-wave consumer-producer) GEMM

### PR DESCRIPTION
**Pull request by ramjana**
-Support for wavelet programming model.

### CODE REVIEW

**Comment by asroy** [comment]
I will resolve the conflict on this PR and polish the code after some urgent tasks

**Comment by aosewski** [comment]
@asroy @zjing14 @rosenrodt  The PR is ready for review.

**Comment by aosewski** [comment]
CI got following error: 
> fatal error: error in backend: Error while trying to spill VGPR0 from class VGPR_32: Cannot scavenge register without an 
> emergency spill slot!

[http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2Fcomposable_kernel/detail/wavelet_model/9/pipeline#step-146-log-340](https://clicktime.symantec.com/15tpJ8xKtrNHYacPWpEg9?h=yhzBN6878-ZWxBE-6d_WTGm2I9KJHP2t7Gxmf2xhDSM=&u=http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%252Fcomposable_kernel/detail/wavelet_model/9/pipeline%23step-146-log-340)

**Comment by aosewski** [comment]
Waiting for PR #516 to get into develop to fix CI.

**Comment by aosewski** [comment]
@asroy  The PR is ready for review

File: include/ck/tensor_operation/gpu/device/impl/device_gemm_multiple_d_xdl_cshuffle.hpp [hunk 1]
```diff
@@ -471,6 +468,8 @@ struct DeviceGemmMultipleD_Xdl_CShuffle : public DeviceGemmMultipleD<ALayout,
                                               arg.block_2_etile_map_);
             };
 
+            const auto K = arg.a_grid_desc_m_k_.GetLength(I1);
```
Comment (asroy) [inline include/ck/tensor_operation/gpu/device/impl/device_gemm_multiple_d_xdl_cshuffle.hpp:471]: This is a bug fix?

File: include/ck/tensor_operation/gpu/grid/block_to_ctile_map.hpp [hunk 2]
```diff
@@ -154,6 +154,50 @@ struct BlockToCTileMap_M00_N0_M01Adapt
         index_t idx_M01          = idx_M0 % M01_;
         index_t idx_N0_M01_local = idx_N0 + idx_M01 * N0;
 
+        /**
```
Comment (asroy) [inline include/ck/tensor_operation/gpu/grid/block_to_ctile_map.hpp:157]: Nice!

**Review by asroy — APPROVED** [review]

File: include/ck/tensor_operation/gpu/device/impl/device_gemm_multiple_d_xdl_cshuffle.hpp (hunk 1 above)
Comment (aosewski) [reply include/ck/tensor_operation/gpu/device/impl/device_gemm_multiple_d_xdl_cshuffle.hpp:471]: This is not my change as for git-blame :P

**Pull request merged**